### PR TITLE
Pointer value improvements

### PR DESCRIPTION
**Pull request by ehaas**
Allow array-to-pointer decay to produce a pointer value
Handle address-of locals as well as globals
Rename "reloc" to pointer for consistency

### CODE REVIEW

File: test/cases/ast/_Float16.c [hunk 1]
```diff
@@ -30,7 +30,7 @@ fn_def: 'fn (x: int, ...) void'
     builtin_call_expr: 'void'
      name: __builtin_va_start
      args:
-      implicit_cast: (array_to_pointer) 'va_list': '*d[1]struct __va_list_tag'
+      implicit_cast: (array_to_pointer) 'va_list': '*d[1]struct __va_list_tag' (value: )
```
Comment (Vexu) [inline test/cases/ast/_Float16.c:33]: Would be nice if pointer values printed the offset (if non-zero).

**Review by Vexu — APPROVED** [review]

File: test/cases/ast/_Float16.c (hunk 1 above)
Comment (ehaas) [reply test/cases/ast/_Float16.c:33]: I meant to ask about this - should I just print the offset (without the name of the variable)?

Comment (Vexu) [reply test/cases/ast/_Float16.c:33]: Printing the `NodeIndex` probably won't be too useful so I guess just the offset is good for now. Maybe add something to distinguish it from just ints.

Comment (ehaas) [reply test/cases/ast/_Float16.c:33]: What if we change the return type of `Value.print` from `@TypeOf(w).Error!void` to something like `@TypeOf(w).Error!?NestedPrint` where `NestedPrint` is
```zig
const NestedPrint = union(enum) {
    pointer: struct {
        node: u32,
        offset: Value,
    },
};
```

**Pull request merged**